editor.planx.uk/src/components/Feedback.tsx

```diff
@@ -253,8 +400,27 @@ const FeedbackComponent: React.FC = () => {
           </FeedbackRow>
         </Container>
       </FeedbackWrapper>
-    </>
-  );
+    );
+  }
+
+  function Feedback(): FCReturn {
```

Comment:
In TypeScript switch statements don't read types as nicely as they should unfortunately. 

I'd suggest adding an exhaustive check here as it would throw a type error if we added a new `FeedbackView` but didn't update this case statement.

Reply:
Tbh maybe we should enforce this with something like https://typescript-eslint.io/rules/switch-exhaustiveness-check/ to avoid having to repeatedly use an exhausive guard. Maybe a nicer solution!

Reply:
Thanks for taking a look at this!

Locally on your branch this isn't quite working for me (it's maybe a me issue!).

If I drop the final "thanks" case from this switch locally I can run `pnpm check` and it picks up an issue with this file - 

```sh
➜  editor.planx.uk git:(mh/setup-feedback-form-conditional-rendering) pnpm check

> editor.planx.uk@ check /Users/dafyddllyr/Code/planx-new/editor.planx.uk
> tsc --noEmit && pnpm lint

src/components/Feedback.tsx:363:24 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.

363   function Feedback(): FCReturn {
                           ~~~~~~~~

Found 1 error in src/components/Feedback.tsx:363
```


However my IDE doesn't pick up the affected line, it just gives an issue right at the top of the affected file - 
![image](https://github.com/theopensystemslab/planx-new/assets/20502206/c4d62809-a9dd-4ccb-9f71-61a825ddeb2b)

(The location of the `.tsconfig.json` should be `/Users/dafyddllyr/Code/planx-new/editor.planx.uk/tsconfig.json`)

Are you hitting these issues?

Reply:
> Thanks for taking a look at this!
> 
> Locally on your branch this isn't quite working for me (it's maybe a me issue!).
> 
> If I drop the final "thanks" case from this switch locally I can run `pnpm check` and it picks up an issue with this file -
> 
> ```shell
> ➜  editor.planx.uk git:(mh/setup-feedback-form-conditional-rendering) pnpm check
> 
> > editor.planx.uk@ check /Users/dafyddllyr/Code/planx-new/editor.planx.uk
> > tsc --noEmit && pnpm lint
> 
> src/components/Feedback.tsx:363:24 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.
> 
> 363   function Feedback(): FCReturn {
>                            ~~~~~~~~
> 
> Found 1 error in src/components/Feedback.tsx:363
> ```
> 
> However my IDE doesn't pick up the affected line, it just gives an issue right at the top of the affected file - ![image](https://private-user-images.githubusercontent.com/20502206/299312166-c4d62809-a9dd-4ccb-9f71-61a825ddeb2b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYxMDExMTUsIm5iZiI6MTcwNjEwMDgxNSwicGF0aCI6Ii8yMDUwMjIwNi8yOTkzMTIxNjYtYzRkNjI4MDktYTlkZC00Y2NiLTlmNzEtNjFhODI1ZGRlYjJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI0VDEyNTMzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0MjkxYTY5NGRhNmY2ZWQwZjU3YWU2MWNhYjRhNmI4Y2FhNmJmNjlmYzY0MTRmNzA1YzA4NjMxNjVmYzNkZjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vd4SCICrU42dgPFgyahTY1DhqTsVFdZYSzzSOwUDbaY)
> 
> (The location of the `.tsconfig.json` should be `/Users/dafyddllyr/Code/planx-new/editor.planx.uk/tsconfig.json`)
> 
> Are you hitting these issues?

This is a bit of a weird one 🤔 

Normally I use `npx lint-staged` before I make a commit and when I remove the `thanks` case I see:

```
❯ npx lint-staged
✔ Preparing lint-staged...
❯ Running tasks for staged files...
  ❯ package.json — 1 file
    ❯ src/**/*.{js,jsx,ts,tsx} — 1 file
      ✖ eslint --fix [FAILED]
    ✔ src/**/*.{js,jsx,ts,tsx,json,css,scss,md,html} — 1 file
    ❯ src/**/*.{ts,tsx} — 1 file
      ✖ scripts/tsc-lint.sh [KILLED]
↓ Skipped because of errors from tasks. [SKIPPED]
✔ Reverting to original state because of errors...
✔ Cleaning up temporary files...

✖ eslint --fix:

/Users/mike/planx-new/editor.planx.uk/src/components/Feedback.tsx
   95:40  warning  Unexpected any. Specify a different type                                             @typescript-eslint/no-explicit-any
   98:28  warning  Unexpected any. Specify a different type                                             @typescript-eslint/no-explicit-any
  345:12  warning  'ThanksForFeedback' is defined but never used. Allowed unused vars must match /^_/u  @typescript-eslint/no-unused-vars
  364:13  error    Switch is not exhaustive. Cases not matched: "thanks"                                @typescript-eslint/switch-exhaustiveness-check

✖ 4 problems (1 error, 3 warnings)


✖ scripts/tsc-lint.sh failed without output (KILLED).
```

When I try `pnpm check` I see:

```
Checking formatting...
All matched files use Prettier code style!
❯ pnpm check

> editor.planx.uk@ check /Users/mike/planx-new/editor.planx.uk
> tsc --noEmit && pnpm lint

src/components/Feedback.tsx:363:24 - error TS2366: Function lacks ending return statement and return type does not include 'undefined'.

363   function Feedback(): FCReturn {
                           ~~~~~~~~


Found 1 error in src/components/Feedback.tsx:363
```

In both cases I don't see any warning in my editor 🤔

Reply:
If you have time maybe we could jump on a huddle at @DafyddLlyr as I'm slightly perplexed by this one. I think my setup/workflow hasn't been quite right.

Reply:
Yeah lets do that 👍 

Could you please drop these changes from the PR so that we can look at it in isolation, and don't hold up this PR unnecessarily?

Reply:
I'm also thinking `FCReturn` might actually be correct as if a `case` is missing i.e. the `currentFeedbackView` is `"thanks"` then the component does indeed lack a return statement and isn't guaranteed to return a react component?